lib/pem.ml

```diff
@@ -54,24 +54,27 @@ module Cs = struct
       | `Data cs :: tail -> accumulate t (cs :: acc) tail
       | `End t' :: tail ->
         if String.equal t t' then
-          Ok (concat "" (List.rev acc), tail)
+          let data = match Base64.decode (concat "" (List.rev acc)) with
+            | Ok data -> Ok (t, data)
+            | Error e -> Error e
+          in
+          data, tail
         else
-          Error (`Msg ("invalid end, expected " ^ t ^ ", found " ^ t'))
-      | _ :: _ -> Error (`Msg "invalid line, expected data or end")
-      | [] -> Error (`Msg "end of input")
+          Error (`Msg ("invalid end, expected " ^ t ^ ", found " ^ t')), tail
+      | _ :: tail -> Error (`Msg "invalid line, expected data or end"), tail
+      | [] -> invalid_arg "empty list"
```

Comment:
would you mind to explain why this should be an invalid_arg now?

Reply:
Thanks for catching this, it shouldn't!